contracts/puppeteer/src/contract.rs

```diff
@@ -153,26 +140,67 @@ fn query_kv_query_ids(deps: Deps<NeutronQuery>) -> ContractResult<Binary> {
 }
 
 fn query_delegations(deps: Deps<NeutronQuery>) -> ContractResult<Binary> {
-    let data = DELEGATIONS_AND_BALANCE.load(deps.storage)?;
-    to_json_binary(&DelegationsResponse {
-        delegations: data.data.delegations,
-        remote_height: data.remote_height,
-        local_height: data.local_height,
-        timestamp: data.timestamp,
-    })
+    let puppeteer_base = Puppeteer::default();
+    match puppeteer_base
+        .last_complete_delegations_and_balances_key
+        .may_load(deps.storage)?
+    {
+        None => to_json_binary(&DelegationsResponse {
+            delegations: Delegations {
+                delegations: vec![],
+            },
+            remote_height: 0,
+            local_height: 0,
+            timestamp: Timestamp::default(),
+        }),
+        Some(last_key) => {
+            let last_data = puppeteer_base
+                .delegations_and_balances
+                .may_load(deps.storage, &last_key)?
```

Comment:
Why is it may_load here? If there is a key for data, shouldn't we expect this data to be always there?

Reply:
initial state, when we don't have any complete set of answers

Reply:
but in general it seems we can just `load` here. wdyt?

Reply:
yes, that's what I am thinking, just using load() should be sufficient

Reply:
well, from core pow, we need these queries work even at start wo any complete set or even delegations. This way let's keep may_load with returning empty struct

Reply:
You are using .ok_or(Error) right at the next line anyway :^)
Might as well just switch to using load()